hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -177,8 +177,14 @@ private Account.Builder updateBuilder(
         if (op.hasDeclineReward()) {
             builder.declineReward(op.declineReward().booleanValue());
         }
-        if (op.hasStakedAccountId() || op.hasStakedNodeId()) {
-            builder.stakedNumber(getStakedId(op.stakedId().kind().toString(), op.stakedNodeId(), op.stakedAccountId()));
+        if (op.hasStakedAccountId()) {
+            if (op.stakedAccountId() != null && 0 == op.stakedAccountId().accountNum()) {
```

Comment:
I think `hasStakedAccountId()` checks for null already

Reply:
Yes, removed it.